content/posts/jenkins-x-flow-integration.md

```diff
@@ -0,0 +1,147 @@
+
+In today’s heterogenous technology environment, most organizations tend to have a mix of modern cloud native architectures as well as more traditional workloads which get deployed either on-prem or within the cloud. In the latter case, a combination of Jenkins X (performing CI steps) and CloudBees Flow (handling the deployment) can add a huge amount of flexibility and power to a Continuous Delivery process.  Jenkins X can be easily extended to accommodate any type of workload required - it can be a full end to end CI/CD tool for building, deploying, and running applications all within a Kubernetes cluster, or it can handle CI while offloading other release and deployment tasks to another solution.  In this blog post we’re going to cover how Jenkins X can be extended to offload release/deployment tasks to [CloudBees Flow](https://www.cloudbees.com/cloudbees-acquires-electric-cloud).
+
+# Extending Jenkins X
+For the purposes of this blog, we’re going to be focusing on the Jenkins X serverless pipeline execution engine with Tekton (See https://jenkins-x.io/architecture/jenkins-x-pipelines/). There are two main ways to customize a Jenkins X pipeline in order to integrate with CloudBees Flow.  The first and simplest would be to modify the jenkins-x.yml (more information on Jenkins X pipelines: https://jenkins-x.io/architecture/jenkins-x-pipelines/#differences-to-jenkins-pipelines and the jenkins-x.yml file) pipeline file in the source code repo for the project we’re going to build.  The other way is to extend the [Jenkins X build packs](https://jenkins-x.io/architecture/build-packs/) and modify the build pack for the language/build tool you want to use.  Both will work, but by forking the build packs you can get reuse across multiple projects using the build pack you extend. In this example, we’ll walk through how to extend the Jenkins X build packs.
+
+## Creating our Cluster and Installing Jenkins X
+To start, we’ll fork the Jenkins X Kubernetes build packs into our own repository: https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes.  Later we'll be extending the maven build pack to support a REST API call into CloudBees Flow.
+
+Then it’s time to create a Kubernetes cluster on GKE using Jenkins X and [Tekton](https://github.com/tektoncd/pipeline): 
+
+```bash
+jx create cluster gke --tekton --no-tiller
+```
+
+Fill out the options.  For example:
+
+{{< image src="/img/jenkins-x-flow-integration/jx-create-cluster.jpg" alt="Jenkins X create cluster output" >}}
```

Comment:
These screenshots are kind of hard to read - would it be that difficult to add as code blocks? Perhaps as bash or shell - see https://prismjs.com/#languages-list

Reply:
Good idea, converting these to shell